components/Form/index.js

```diff
@@ -0,0 +1,332 @@
+import styled from "styled-components";
+import { css } from "styled-components";
+import { useRouter } from "next/router";
+import { useState } from "react";
+
+export default function Form({ onSubmit }) {
+  const [sleep, setSleep] = useState(true);
+  const [hygiene, setHygiene] = useState(false);
+  const [medicine, setMedicine] = useState(false);
```

Comment:
You can achieve the same by using only one state. In this state you will save the category as a string (e.g. "sleep", "medicine"...). If you want the details of the first category to be shown on initial render of the form, you can do so by setting this category as initial value of your state:

```suggestion
const [category, setCategory] = useState("sleep")
```
When the user selects a different category, you write the category name into your state:

```jsx
<StyledInput
                  onClick={() => {
                    setCategory("medicine");
                  }}
...
/>
```
In each `StyledSection` you just check if the respective category is saved in your state. So you end up with a `true` or `false` for `showDetails`.

```jsx
<StyledSection showDetails={isShown === "sleep"}>
```

The styles can then be changed accordingly:

```js
  ${({ showDetails }) => (showDetails ? "size: 20px;" : "display: none;")}
```

As you only want to change one property, you do not need the `css` function from styled-components.